plugin/storage/es/factory.go

```diff
@@ -109,9 +109,7 @@ func (f *Factory) CreateSpanWriter() (spanstore.Writer, error) {
 
 // CreateDependencyReader implements storage.Factory
 func (f *Factory) CreateDependencyReader() (dependencystore.Reader, error) {
-	reader := esDepStore.NewDependencyStore(f.primaryClient, f.logger, f.primaryConfig.GetIndexPrefix(),
-		f.primaryConfig.GetIndexDateLayoutDependencies(), f.primaryConfig.GetMaxDocCount())
-	return reader, nil
+	return createDependencyReader(f.logger, f.primaryClient, f.primaryConfig)
```

Comment:
nit: I think we can just `return esDepStore.NewDependencyStore(...)`. 

I can see the intention here is to maintain consistency, though I think `CreateSpanReader/Writer` extract the function for reuse with its `Archive` equivalent. The `DependencyStore` doesn't support archiving AFAIK, so there doesn't seem to be much benefit of extracting the function for the cost of the additional indirection, unless you had other reasons?

What do you think?

Reply:
While I understand you point, I am actually in favor of consistency. And there certainly might arise the need to add archiving, and last but not least, the compiler will optimize the additional call away.

In short, if you insist I will certainly change this, but I believe the code is just more approachable such similar things, as in creating readers for three types of stores, are not handled differently.

Reply:
We can leave this as is, I'm not strongly attached to my suggestion. 😄